core/lib/src/rocket.rs

```diff
@@ -468,6 +469,61 @@ impl Rocket<Build> {
         self
     }
 
+    /// Modify previously attached fairings. This should typically be used by on_ignite Fairings,
+    /// to register new handlers or otherwise enable Fairings to work together.
+    ///
+    /// ```rust,no_run
+    /// # #[macro_use] extern crate rocket;
+    /// # extern crate rocket_dyn_templates;
+    /// use rocket::Rocket;
+    /// use rocket::fairing::AdHoc;
+    /// use rocket_dyn_templates::Templates;
+    ///
+    /// #[launch]
+    /// fn rocket() -> _ {
+    ///     rocket::build()
+    ///         .attach(Templates::default())
+    ///         .attach(AdHoc::on_ignite("Register new template function", |r| {
+    ///             r.modify_fairing(|f: &mut Templates| {
+    ///                 f.register_function("some_function", some_function)
+    ///             })
+    ///         })))
+    /// }
+    /// ```
+    ///
+    /// If multiple instances of the same type of Fairing have been attached, the modify function
+    /// will be run on each instance.
+    #[must_use]
+    pub fn modify_fairings<F: Fairing, E: fmt::Debug>(mut self, modifier: impl Fn(&mut F) -> Result<(), E>) -> Result<Self, Self> {
```

Comment:
Could this be the following?

```rust
    pub fn modify_fairings<F: Fairing>(mut self, modifier: impl FnMut(&mut F)) -> Self;
```

This is much simpler to reason about and implement, though perhaps I'm missing something that we gain with the other signature.